chain/src/txhashset.rs

```diff
@@ -211,15 +211,24 @@ impl TxHashSet {
 
 	/// Compact the MMR data files and flush the rm logs
 	pub fn compact(&mut self) -> Result<(), Error> {
-		let horizon = global::cut_through_horizon();
 		let commit_index = self.commit_index.clone();
+		let head = commit_index.head()?;
+		let current_height = head.height;
+
+		// horizon for compacting is based on current_height
+		let horizon = (current_height as u32).saturating_sub(global::cut_through_horizon());
```

Comment:
We were using the `cut_through_horizon` effectively as a hardcoded block height.
We need to look at the current height and offset by `cut_through_horizon` blocks in the past.

Reply:
Oopsie, good catch.